ui/dashboard.js

```diff
@@ -78,10 +78,32 @@ function create_box (id, icon, number, currency_icon) {
     settings.page_index = $('#page-wrapper').html();
 }
 
+function update_eth_node_conn_status_ui(is_local_eth) {
+    let str;
+    if (!is_local_eth) {
+        str='<a class="eth-node-status-link" href="#"><i class="fa fa-unlink fa-fw"></i></a><ul class="dropdown-menu dropdown-eth-status alert-warning"><p>Not connected to a local ethereum node</p></ul>';
+    } else {
+        str='<a class="eth-node-status-link" href="#"><i class="fa fa-link fa-fw"></i></a><ul class="dropdown-menu dropdown-eth-status alert-success"><p>Connected to a local ethereum node</p></ul>';
+    }
+    $('.eth-node-status').html(str);
+}
 
 
 var alert_id = 0;
 function add_alert_dropdown(alert_text, alert_time) {
+    if (
+        alert_text.indexOf("ethereum") > 0
```

Comment:
Why have 2 different checks here for the same thing? Also having them all inside the `if()` makes it harder to read. A nice way to improve readability here would be:

```javascript
let connection_fail = alert_text.indexOf('Could not connect to a local ethereum node') >= 0;
let no_sync = alert_text.indexOf('Could not transact with/call contract function') >= 0;
if (connection_faill || no_sync) {
    update_eth_node_connection_status_ui(false);
} else {
    update_eth_node_connection_status_ui(true);
}
```